### ISSUE TITLE
Unable to find container images

### ISSUE THREAD

**Issue by timbrd** [bitnami-labs/sealed-secrets#822]
I am not able to install sealed-secrets anymore. Using helm chart (v2.1.5), my k8s cluster is not able to find the container image:

```
13s         Normal    Pulling                       pod/sealed-secrets-controller-5d8db499cd-zthst    Pulling image "quay.io/bitnami/sealed-secrets-controller:v0.17.4"
12s         Warning   Failed                        pod/sealed-secrets-controller-5d8db499cd-zthst    Failed to pull image "quay.io/bitnami/sealed-secrets-controller:v0.17.4": rpc error: code = Unknown desc = reading manifest v0.17.4 in quay.io/bitnami/sealed-secrets-controller: unauthorized: access to the requested resource is not authorized
```

Shouldn't there be a repository at https://quay.io/repository/bitnami/sealed-secrets-controller ?

**Comment by mhulscher** [comment]
Looks like someone made the organization private, or purged all images 🤷🏾‍♂️

**Comment by Nicarim** [comment]
Seeing same issue (bumping for visibility). Can't proceed with deployments with that missing.

**Comment by jeanguirro** [comment]
Having the same issue:
```shell
> docker pull quay.io/bitnami/sealed-secrets-controller:v0.16.0
Error response from daemon: unauthorized: access to the requested resource is not authorized
```

**Comment by radoslav-tomov** [comment]
+1

**Comment by jackcaldwell** [comment]
Facing same issue here.

```
Failed to pull image "quay.io/bitnami/sealed-secrets-controller:v0.13.1": rpc error: code = Unknown desc = Error response from daemon: unauthorized: access to the requested resource is not authorized
```

**Comment by mprusinski** [comment]
+1

**Comment by RobMokkink** [comment]
Have the same issue, please fix it.

**Comment by Jc2k** [comment]
There seems to be a mirror here - https://quay.io/repository/56401deb-7290-4d8b-8731-baf6d8b5d27d/sealed-secrets-controller?tab=tags. The hash for 0.17.4 matches what I have locally (f4bc4aca1ae011b89e0e6b997f63a89c8cf464cd41c0bd4544218878200b9b69).

**Comment by pigfrown** [comment]
+1 .. anyone know how to build the docker image from this repo? the Dockerfile doesn't build (files in dist/ are missing)

**Comment by arnaud-mey** [comment]
@pigfrown you need to first run `make controller`

**Comment by LPby** [comment]
+1
```
$ docker pull quay.io/bitnami/sealed-secrets-controller:v0.17.4
Error response from daemon: unauthorized: access to the requested resource is not authorized

```

**Comment by Jc2k** [comment]
https://github.com/bitnami-labs/sealed-secrets/pull/824 - prepare for a new release on a different repo I guess.

**Comment by timbrd** [comment]
> #824 - prepare for a new release on a different repo I guess.

Awesome. How about switching to a new registry, but keeping the images at the old one for some time?

**Comment by LPby** [comment]
any estimation when image will be available at dockerhub?

**Comment by timbrd** [comment]
The new docker image, tagged as v0.17.5 and uploaded to [dockerhub](https://hub.docker.com/r/bitnami/sealed-secrets-controller/tags), works for me .

**Comment by agarcia-oss** [comment]
Hi everyone, 

The new controller image is already available from the new oficial DockerHub repository:
https://hub.docker.com/r/bitnami/sealed-secrets-controller/tags

The reason for this change is to use our [Dockerhub Bitnami](https://hub.docker.com/u/bitnami)account as a distribution platform of the controller image. Doing so we can leverage in the advantages of being a [verified publisher](https://docs.docker.com/docker-hub/publish/#:~:text=The%20Docker%20Verified%20and%20Publisher,components%20to%20build%20new%20applications.), such as not been affected by dockerhub pull rate limits. 

Unfortunately, quay.io could not provide that level of support for our images and we were forced to migrate the images.

Please accept our apologies if this action has created some disturbances during the process. 
Regards

**Comment by DionJones615** [comment]
> The new controller image is already available from the new oficial DockerHub repository: https://hub.docker.com/r/bitnami/sealed-secrets-controller/tags

Older images that were on quay.io are not accessible on quay or dockerhub right now, impacting our production systems.  Where can we reference older versions?

**Comment by agarcia-oss** [comment]
We are investigating the best way to publish older versions of the controller in the new [DockerHub](https://hub.docker.com/r/bitnami/sealed-secrets-controller/tags) repository.

However, we fully recommend upgrading your clusters to the newest release ([v.0.17.5](https://github.com/bitnami-labs/sealed-secrets/releases/tag/v0.17.5)) Upgrading should be quite straightforward and older versions of sealed secrets will not have our [latest improvements and bugfixes](https://github.com/bitnami-labs/sealed-secrets/blob/main/RELEASE-NOTES.md).

**Comment by abebeos** [comment]
@agarcia-oss , @all: i have this config

```yaml
---
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: HelmRepository
metadata:
  name: sealed-secrets
  namespace: flux-system
spec:
  interval: 1h0m0s
  url: https://bitnami-labs.github.io/sealed-secrets
```

0.16.0 fails, but changing to 0.17.5 seems to fail too.

Anything else needs to be changed?

**Comment by rouke-broersma** [comment]
> 

The chart version for `0.17.5` is `2.1.6` and not `0.17.5`

**Comment by abebeos** [comment]
@rouke-broersma , all should be good now, they have updated the helm-charts:

https://github.com/bitnami-labs/sealed-secrets/commit/e68de2c17df40a22192b860ece47d726973ead85#diff-c293662387deb9efc2693cf116c5b930beaf110e9d6e5074c985ba6c8311e313

**Comment by rouke-broersma** [comment]
> @rouke-broersma , all should be good now, they have updated the helm-charts:
> 
> [e68de2c#diff-c293662387deb9efc2693cf116c5b930beaf110e9d6e5074c985ba6c8311e313](https://github.com/bitnami-labs/sealed-secrets/commit/e68de2c17df40a22192b860ece47d726973ead85#diff-c293662387deb9efc2693cf116c5b930beaf110e9d6e5074c985ba6c8311e313)

Um yes, I know. I was replying to your comment saying 

>0.16.0 fails, but changing to 0.17.5 seems to fail too.
Anything else needs to be changed?

This was fixed last night and you asked 2 hours ago.

**Comment by abebeos** [comment]
> This was fixed last night and you asked 2 hours ago.

yes, tested last night, commented 2 hours ago, then saw the commit (which should make 0.17.5 work again, though 0.16.0 possibly still fails, will check later). All good.

**Comment by abebeos** [comment]
> The chart version for `0.17.5` is `2.1.6` and not `0.17.5`

One momennt, are you telling me that in order to get version 0.17.5, i need to fetch chart-version 2.1.6 ?

**Comment by rouke-broersma** [comment]
> > The chart version for `0.17.5` is `2.1.6` and not `0.17.5`
> 
> One momennt, are you telling me that in order to get version 0.17.5, i need to fetch chart-version 2.1.6 ?

Yes, the chart version that includes the 0.17.5 docker hub images change is version 2.1.6.

**Comment by abebeos** [comment]
Oh, I see, thank you, you really saved me here.

Confirming that a usual fluxcd setup works by updating to 0.17.5 (via 2.1.6 chart)

(though... the amount of mental indirections one need to survive modern IT...)

Are you aware if the 0.16.0 version gets a backport, thus existing builds can be 'revived' without an update to 0.17.5(via 2.1.6)?

**Comment by rouke-broersma** [comment]
I don't know, they said in this thread that they're looking into it.

**Comment by agarcia-oss** [comment]
Hi everyone, 

A quick update on this task: we have uploaded the sealed-secrets controller images from previous versions in our [Dockerhub oficial repository](https://hub.docker.com/r/bitnami/sealed-secrets-controller/tags). The full list of available versions is:
```
v0.9.0
v0.9.1
v0.9.2
v0.9.3
v0.9.4
v0.9.5
v0.9.6
v0.9.7
v0.9.8
v0.10.0
v0.11.0
v0.12.0
v0.12.1
v0.12.2
v0.12.3
v0.12.4
v0.12.5
v0.12.6
v0.13.0
v0.13.1
v0.14.0
v0.14.1
v0.15.0
v0.16.0
v0.17.0
v0.17.3
v0.17.4
v0.17.5
```
This covers versions that older that 2,5 years (and, again, we cannot recommend the use of releases so outdated). As a bonus, the controller binary has being compiled with our latest supported go version, which means a more efficient binary on most cases.  

Restoring the controller is just a matter of updating the chart value accordingly:
```yaml
image:
  registry: docker.io
  repository: bitnami/sealed-secrets-controller
```

We will pin this issue for future reference and mark it as closed in the following days. 

Regards!

**Comment by alvneiayu** [comment]
hi @abebeos 

I was looking for this old version ([0.16.1](https://github.com/bitnami-labs/sealed-secrets/releases?q=0.16.1&expanded=true)) but I can not find a release of this. Is it possible that you are talking about the helm chart version 1.16.1(using the image 0.16.0)?

Thanks a lot

Álvaro

**Comment by abebeos** [comment]
@alvneiayu , yes, now that I'm wiser, I know that I meant "helm chart version 0.16.1"

**Comment by abebeos** [comment]
@agarcia-oss , maybe edit the original issue text and add something like this to the end (thus people find it immediately) :

## Solution

See https://github.com/bitnami-labs/sealed-secrets/issues/822#issuecomment-1105307200

**Comment by abebeos** [comment]
> This Issue has been automatically marked as "stale" because it has not had recent activity (for 15 days). It will be closed if no further activity occurs. Thanks for the feedback.

Fascinating logic...

**Comment by rufreakde** [comment]
Are the helm charts also updated now to reuse the new image versions?

> agarcia-oss

@agarcia-oss if the charts are updated to leverage the newest image versions then we would not need to update to a new chart using this versions. https://bitnami-labs.github.io/sealed-secrets/index.yaml 

From my side we are using chart version: 2.0.1

**Comment by agarcia-oss** [comment]
Hi everyone, 

We can close this issue, since it has been open for several weeks and we have addressed it by releasing our images in the dockerhub oficial repository.

Regards

**Comment by rosenbjerg** [comment]
It seems bitnami has completely dropped quay.io support based on the response here:
[https://community.bitnami.com/t/repositories-no-longer-available-at-quay-io/105179/3](https://community.bitnami.com/t/repositories-no-longer-available-at-quay-io/105179/3)

It may be related to this change:
[https://cloud.redhat.com/blog/quay.io-will-require-a-red-hat-account-after-june-30-2021...wait-what](https://cloud.redhat.com/blog/quay.io-will-require-a-red-hat-account-after-june-30-2021...wait-what)

So yeah, switching to docker hub or GCR seems to be the migration path

**Comment by bit-herder** [comment]
this is super annoying. the old release should work. its one minor version down from this. People's entire clusters depend on sealed secrets working, and new "functionality" in a minor release should not break every other release. Why not just keep the old repo AND use the new one?

**Comment by agarcia-oss** [comment]
Hi @zoharaharoni, 
We will definitely recommend an upgrade to the latest Sealed Secrets version (0.18.0). The version you're running on your cluster is [more than 5 years old](https://github.com/bitnami-labs/sealed-secrets/releases/tag/v0.7.0) and it'll be difficult for us to provide support or assistance for such version.

Out of curiosity, what K8S version are you using there?
